wire-grpc-client/src/main/java/com/squareup/wire/GrpcMethod.kt

```diff
@@ -211,6 +220,154 @@ internal sealed class GrpcMethod<S : Any, R : Any>(
     }
   }
 
+  class GrpcCall<S : Any, R : Any>(
+    path: String,
+    requestAdapter: ProtoAdapter<S>,
+    responseAdapter: ProtoAdapter<R>
+  ) : GrpcMethod<S, R>(path, requestAdapter, responseAdapter) {
+    override fun invoke(
+      grpcClient: GrpcClient,
+      args: Array<Any>
+    ): com.squareup.wire.GrpcCall<S, R> {
+      return object : com.squareup.wire.GrpcCall<S, R> {
+        // TODO(benoit) We cannot eagerly create it because we're waiting on the request.
+        private var call: Call? = null
+
+        // TODO(benoit) How do we deal with `Call` being lazily instantiated?
```

Comment:
Throw `IllegalStateException` if someone calls `timeout` before executing the call?

Reply:
Nah, you can create a `Timeout` instance that forwards.

Reply:
You're supposed to be able to invoke it before you execute the call. If you create your own Timeout you can merge it with the Call at the time of the Call's creation.

Reply:
```
class LateInitTimeout : ForwardingTimeout(Timeout()) {
  fun init(newDelegate: Timeout) {
    val oldDelegate = this.delegate
    newDelegate.timeout(oldDelegate.timeoutNanos(), TimeUnit.NANOSECONDS)
    if (oldDelegate.hasDeadline()) newDelegate.deadlineNanoTime(oldDelegate.deadlineNanoTime())
    delegate = newDelegate
  }
}
```

Reply:
And so how about cancellation? Isn't odd that `GrpcStreaming` cancel before client didn't do anything but `GrpcCall` doesn't really?